src/components/translation-builder/translation-builder.component.tsx

```diff
@@ -0,0 +1,133 @@
+        updatedSchema.translations = {};
+      }
+      updatedSchema.translations[langCode] = updatedTranslations;
+      onUpdateSchema(updatedSchema);
+    }
+  };
+
+  return (
+    <div className={styles.translationBuilderContainer}>
+      <div className={styles.translationBuilderHeader}>
+        <Tabs>
+          <TabList aria-label="Form previews">
+            <Tab>{t('all', 'All')}</Tab>
+            <Tab>{t('translated', 'Translated')}</Tab>
+            <Tab>{t('untranslated', 'Untranslated')}</Tab>
+          </TabList>
+        </Tabs>
+
+        <div className={styles.languageTools}>
+          <div className={styles.languagePath}>
+            <span className={styles.language}>English (en)</span>
+            <ArrowRight className={styles.arrow} />
+          </div>
+
+          <Dropdown
+            id="target-language"
+            items={languageOptions}
+            itemToString={(item) => item?.label ?? ''}
+            label={t('selectLanguage', 'Select language')}
+            titleText=""
+            selectedItem={languageOptions.find((opt) => opt.code === selectedLanguageCode)}
+            onChange={({ selectedItem }) => {
+              if (selectedItem) setSelectedLanguageCode(selectedItem.code);
+            }}
+          />
+
+          <IconButton kind="ghost" label={t('downloadTranslation', 'Download translation')} size="md">
+            <Download />
+          </IconButton>
+        </div>
+      </div>
+
+      {loading ? (
+        <InlineLoading description={t('loadingTranslations', 'Loading translations...')} />
+      ) : error ? (
+        <InlineNotification kind="error" title={t('error', 'Error')} subtitle={error} lowContrast />
+      ) : (
+        <div className={styles.translationEditor}>
+          {Object.entries(translations).length > 0 ? (
+            Object.entries(translations).map(([key, value]) => (
+              <div key={key} className={styles.translationRow}>
+                <div className={styles.translationKey}>{key}</div>
+                <div className={styles.translatedKey}>{value}</div>
+                <div className={styles.inlineControls}>
+                  <IconButton
+                    kind="ghost"
+                    label={t('editString', 'Edit string')}
+                    onClick={() => {
+                      const editingKey = key;
+                      const dispose = showModal('edit-translation-modal', {
+                        onClose: () => dispose(),
+                        originalKey: editingKey,
+                        initialValue: translations[editingKey],
+                        onSave: (newValue: string) => {
+                          const updated = { ...translations, [editingKey]: newValue };
+                          setTranslations(updated);
+                          handleUpdateValue(editingKey, newValue);
+                          dispose();
+                        },
+                      });
```

Comment:
This function definition should be separate and wrapped in a `useCallback` and then passed on to here